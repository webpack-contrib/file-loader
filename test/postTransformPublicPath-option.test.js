import webpack from './helpers/compiler';

describe('when applied with `postTransformPublicPath` option', () => {
  it('matches snapshot for returned input parameter value without modification', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          postTransformPublicPath: (p) => p,
        },
      },
    };

    const stats = await webpack('fixture.js', config);
    const [module] = stats.toJson().modules;
    const { assets, source } = module;

    expect({ assets, source }).toMatchSnapshot();
  });

  it('matches snapshot for appending to input parameter value', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          postTransformPublicPath: (p) => `${p} + "/test"`,
        },
      },
    };

    const stats = await webpack('fixture.js', config);
    const [module] = stats.toJson().modules;
    const { assets, source } = module;

    expect({ assets, source }).toMatchSnapshot();
  });
});

describe('when applied with `publicPath` and `postTransformPublicPath` option', () => {
  describe('`{String}` value', () => {
    it('matches snapshot for returned input parameter value without modification', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath: 'public_path/',
            postTransformPublicPath: (p) => p,
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { assets, source } = module;

      expect({ assets, source }).toMatchSnapshot();
    });

    it('matches snapshot for appending to input parameter value', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath: 'public_path/',
            postTransformPublicPath: (p) => `${p} + "?test=test"`,
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { assets, source } = module;

      expect({ assets, source }).toMatchSnapshot();
    });

    it('matches snapshot for prefixing with __webpack_public_path__', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath: 'public_path/',
            postTransformPublicPath: (p) => `__webpack_public_path__ + ${p}`,
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { assets, source } = module;

      expect({ assets, source }).toMatchSnapshot();
    });
  });

  describe('`{Function}` value', () => {
    it('matches snapshot for returned input parameter value without modification', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath(url) {
              return `public_path/${url}`;
            },
            postTransformPublicPath: (p) => p,
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { assets, source } = module;

      expect({ assets, source }).toMatchSnapshot();
    });

    it('matches snapshot for appending to input parameter value', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath(url) {
              return `public_path/${url}`;
            },
            postTransformPublicPath: (p) => `${p} + "?test=test"`,
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { assets, source } = module;

      expect({ assets, source }).toMatchSnapshot();
    });

    it('matches snapshot for prefixing with string', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath: 'public_path/',
            postTransformPublicPath: (p) => `"path_prefix/" + ${p}`,
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { assets, source } = module;

      expect({ assets, source }).toMatchSnapshot();
    });

    it('matches snapshot for prefixing with __webpack_public_path__', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath: 'public_path/',
            postTransformPublicPath: (p) => `__webpack_public_path__ + ${p}`,
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { assets, source } = module;

      expect({ assets, source }).toMatchSnapshot();
    });
  });
});
