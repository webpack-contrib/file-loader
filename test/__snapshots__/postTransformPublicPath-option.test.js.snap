// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`postTransformPublicPath\` option matches snapshot for appending to input parameter value 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\" + \\"/test\\";",
}
`;

exports[`when applied with \`postTransformPublicPath\` option matches snapshot for returned input parameter value without modification 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`when applied with \`publicPath\` and \`postTransformPublicPath\` option \`{Function}\` value matches snapshot for appending to input parameter value 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\" + \\"?test=test\\";",
}
`;

exports[`when applied with \`publicPath\` and \`postTransformPublicPath\` option \`{Function}\` value matches snapshot for prefixing with __webpack_public_path__ 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`when applied with \`publicPath\` and \`postTransformPublicPath\` option \`{Function}\` value matches snapshot for prefixing with string 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"path_prefix/\\" + \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`when applied with \`publicPath\` and \`postTransformPublicPath\` option \`{Function}\` value matches snapshot for returned input parameter value without modification 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`when applied with \`publicPath\` and \`postTransformPublicPath\` option \`{String}\` value matches snapshot for appending to input parameter value 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\" + \\"?test=test\\";",
}
`;

exports[`when applied with \`publicPath\` and \`postTransformPublicPath\` option \`{String}\` value matches snapshot for prefixing with __webpack_public_path__ 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`when applied with \`publicPath\` and \`postTransformPublicPath\` option \`{String}\` value matches snapshot for returned input parameter value without modification 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;
